src/main/haskell/kore/src/Kore/Step/Strategy.hs

```diff
@@ -198,35 +203,121 @@ seq a continue === a
 continue :: Strategy prim
 continue = Continue
 
-{- | Run a simple state machine.
+data ExecutionGraph config = ExecutionGraph
+    { root :: Graph.Node
+    , graph :: Gr config ()
+    , history :: [[ConfigNode config]]
+    }
+    deriving(Eq, Show)
+
+-- | A temporary data structure used to construct the @ExecutionGraph@. 
+-- Well, it was intended to be temporary, but for the purpose of making
+-- @pickLongest@ fast it turns out to be useful to keep it around. 
+data ConfigNode config = ConfigNode
+    { config :: config
+    , timestep :: Int
+    -- ^`timestep` represents the time at which a configuration was generated
+    -- during execution. Identical configurations occuring at different times
+    -- are considered different for the purposes of this algorithm. 
+    , nodeId :: Int
+    -- ^ `nodeId` is the hash of `config` and `timestep`.
```

Comment:
Haddock uses single quotes `'` not backticks.

What happens if there is a hash collision?